addons/mrp/models/mrp_workorder.py

```diff
@@ -365,7 +365,7 @@ def _onchange_operation_id(self):
 
     @api.onchange('date_planned_start', 'duration_expected')
     def _onchange_date_planned_start(self):
-        if self.date_planned_start and self.duration_expected:
+        if self.date_planned_start and self.duration_expected and self.workcenter_id:
```

Comment:
You should add `workcenter_id` it in the onchange, no ?

Reply:
Are you asking me because you know it needs to be done, or because you are unsure ?

Personally I am not sure, because the issue at hand here is that there is a traceback because the `self.workcenter_id` is not guarded. 

In the current state of odoo, changing the workcenter_id when there are `self.date_planned_start and self.duration_expected` doesn't trigger this onchange (maybe that's also a bug actually ?)
On the other hand, in the current state of Odoo it is impossible to set the date_planned_start without workcenter_id

So 2 options : 
- Adding workcenter to the onchange and maybe trigger too many times the planning (maybe inconsistencies ?)
- Make those two fields readonly until the workcenter has been selected, to ensure a correct scheduling (maybe the date planner will someday in the future be able to grey out the unavailable date of the work center, so the workcenter will be required even before the onchange)

What do you think @ryv-odoo ? :smile:

Reply:
I was unsure. 
IMO, we should always take in account the calendar resource to avoid inconsistent planning (first option). 
Then even if  the work center is set before the `date_planned_start` it should take in account the resource of the workcenter. (Call onchange too much time is a better idea than not enough in my opinion). Also this solution is transparent for the user in the most of case (the `date_planned_finished` is hide by default).

"Make those two fields readonly until the workcenter has been selected, to ensure a correct scheduling (maybe the date planner will someday in the future be able to grey out the unavailable date of the work center, so the workcenter will be required even before the onchange)" -> Good alternative but it seems not very flexible and confusing for the user.

Then it is ok for you to add the `workcenter_id` in the onchange ?

Reply:
I definitely agree, thanks for taking the time :smile: 
The changes have been pushed, checks are restarting